File: 881-boats-to-save-people/881-boats-to-save-people.py
class Solution:
    def numRescueBoats(self, people: List[int], k: int) -> int:
        
        # try to use two pointer for adding/prefix sum !! 
        
            people.sort()
            l , r = 0 ,len(people)-1
            ant = 0
            while l <=r:
                ant +=1 
                if people[l]+people[r] <=k:
                    l+=1 
                r -=1 
            return ant 

File: 881-boats-to-save-people/881-boats-to-save-people.cpp
class Solution {
public:
    int numRescueBoats(vector<int>& people, int limit) {
        
        int l = 0 , r = people.size()-1;
        int cnt =0;
        sort(people.begin(),people.end());
        while(l <=r){
            cnt ++;
            if( people[l]+people[r]<=limit) l++;
            r--;
        }
        return cnt; 
        
    }
};